tests/test_aliases.py

```diff
@@ -345,9 +344,9 @@ def alias_generator(x):
         'e_generator_parent',
     ]
     assert [f.alias for f in Child.__fields__.values()] == [
-        'a_field_child',
         'b_config_child',
         'c_field_parent',
         'd_config_parent',
         'e_generator_child',
+        'a_field_child',
```

Comment:
Thanks a lot for the quick fix here. I think this behavior is consistent with the dataclasses as well

```python
from dataclasses import dataclass, fields

@dataclass(order=True)
class A:
    a: int
    c: int = 0
    b: int = 1

@dataclass(order=True)
class B(A):
    e: int = 1
    a: int = 1
 
>>> fields(B)
(Field(name='a',type=<class 'int'>,default=1,default_factory=<dataclasses._MISSING_TYPE object at 0x7fedf0188760>,init=True,repr=True,hash=None,compare=True,metadata=mappingproxy({}),_field_type=_FIELD),
 Field(name='c',type=<class 'int'>,default=0,default_factory=<dataclasses._MISSING_TYPE object at 0x7fedf0188760>,init=True,repr=True,hash=None,compare=True,metadata=mappingproxy({}),_field_type=_FIELD),
 Field(name='b',type=<class 'int'>,default=1,default_factory=<dataclasses._MISSING_TYPE object at 0x7fedf0188760>,init=True,repr=True,hash=None,compare=True,metadata=mappingproxy({}),_field_type=_FIELD),
 Field(name='e',type=<class 'int'>,default=1,default_factory=<dataclasses._MISSING_TYPE object at 0x7fedf0188760>,init=True,repr=True,hash=None,compare=True,metadata=mappingproxy({}),_field_type=_FIELD))
```

Just have two comments here

1. If the child has been override here and move to the end, will that affect inherited validators from parent as well because of this order change? 
2. Maybe it is worth adding a inherited child field in the class body directly as another test scenario as well as adding a inherited field that intentionally changed the order to see if the changed order is taken effects

Reply:
The behaviour is actually non consistent with dataclasses but behind the scene it's quite different. Dataclass does not do validation and all annotations are passed to child classes.
This PR was more a showcase to open a discussion than a real proposal (see description of the PR to see why it can't be accepted as is)